### PR TITLE
Create forecast

### PR DESCRIPTION
**Pull request by nagerz**

### CODE REVIEW

File: routes/api/v1/forecast.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
+var express = require('express');
```
Comment (nagerz) [inline routes/api/v1/forecast.js:1]: @plapicola @TyMazey got a nice and long controller function for you. no specific new questions but would be interested if you have any thoughts on refactors/different organization.

File: routes/api/v1/forecast.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            .then(city => {
+              if(city == null){
+                City.create({
+                  city: geodata.city,
+                  state: geodata.state,
+                  lat: geodata.lat,
+                  long: geodata.long,
+                })
+                .then(city => {
+                  Query.create({
```
Comment (allieIsOkay) [inline routes/api/v1/forecast.js:63]: The code here and line 77 can both be pulled in to a function. You might also be able to find a way to remove the need for an else by using a findOrCreate.

File: routes/api/v1/forecast.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+var City = require('../../../models').City;
+var Forecast = require('../../../pojos/forecast');
+
+var pry = require('pryjs');
+const fetch = require('node-fetch');
+//eval(pry.it)
+
+router.get("/", function(req, res, next) {
+  let queryLocation = req.query.location;
+  if (req.body.api_key){
```
Comment (allieIsOkay) [inline routes/api/v1/forecast.js:14]: You might be able to add a function to the user model that would handle the API key lookup and be reusable in other places, which would reduce some of the nesting here.

**Review by allieIsOkay — COMMENTED** [review]
There's some places where you could break your conditionals out in to a function that might make it a little easier to follow. Additionally, it might be worth considering using findOrCreate to remove the need to check if you find a model from a lookup and then make one afterwards in the event you didn't have one.

File: routes/api/v1/forecast.js (hunk 3 above)
Comment (TyMazey) [reply routes/api/v1/forecast.js:14]: Seconded on this

**Review by TyMazey — APPROVED** [review]
I think overall it looks great, the logic is solid, and handles a lot of sad-paths, there is def. a few spots where you could break the logic out into functions and possibly just move all the functions into another file and export them in, but i think for the sake of the project its not worth breaking things and debugging them just to move logic out.

**Pull request merged**